src/observer/ingest_logic.py

```diff
@@ -447,21 +450,114 @@ def download_all_article_metrics():
         lmap(_upsert_metrics_ajson, results)
     return results
 
+# def download_all_article_metrics():
+#    consume.all("metrics/article/summary")
+
+#
+# presspackages
+#
+
+PP_DESC = {
+    'id': [p('id')],
+    'title': [p('title')],
+    'published': [p('published'), todt],
+    'updated': [p('updated', None), todt] # f0114f21 missing an updated date
+}
+
+def _regenerate_presspackage(ppid):
+    "creates PressPackage records with no transaction"
+    data = models.ArticleJSON.objects.get(msid=ppid).ajson
+    mush = render.render_item(PP_DESC, data)
+    return first(create_or_update(models.PressPackage, mush, ['id', 'idstr']))
+
+@transaction.atomic
+def regenerate_presspackage(ppid):
+    return _regenerate_presspackage(ppid)
+
+def regenerate_many_presspackages(ppidlist):
+    return do_all_atomically(_regenerate_presspackage, ppidlist)
+
+def regenerate_all_presspackages():
+    return regenerate_many_presspackages(logic.known_presspackages())
+
+def download_presspackage(ppid):
+    "download a specific press package"
+    return first(consume.single("press-packages/{id}", id=ppid))
+
+def download_all_presspackages():
+    consume.all("press-packages")
+
+
+#
+# profiles
+#
+
+def trunc(len):
+    def _(v):
+        return str(v)[:len]
+    return _
+
+PF_DESC = {
+    'id': [p('id')],
+    # disabled in anticipation of GDPR
+    #'orcid': [p('orcid', None)],
+    #'name': [p('name'), trunc(255)],
+}
+
+def _regenerate_profile(pfid):
+    data = models.ArticleJSON.objects.get(msid=pfid).ajson
```

Comment:
what does a profile have to do with article JSON? 🤔

Reply:
it's transitioning: https://github.com/elifesciences/observer/pull/37/files#diff-a595a69d3c1f7524822efcf3745f60b9R171